prow/plugins/cherrypickunapproved/cherrypick-unapproved.go

```diff
@@ -0,0 +1,141 @@
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+}
+
+type commentPruner interface {
+	PruneComments(shouldPrune func(github.IssueComment) bool)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pr github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.Logger, &pr, pc.CommentPruner, pc.PluginConfig.CherryPickUnapproved.BranchRe, pc.PluginConfig.CherryPickUnapproved.Comment)
+}
+
+func handlePR(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent, cp commentPruner, branchRe *regexp.Regexp, comment string) error {
+	// Only consider the events that indicate opening of the PR and
+	// when the cpApproved and cpUnapproved labels are added or removed
+	cpLabelUpdated := (pr.Action == github.PullRequestActionLabeled || pr.Action == github.PullRequestActionUnlabeled) &&
+		(pr.Label.Name == cpApprovedLabel || pr.Label.Name == cpUnapprovedLabel)
+	if pr.Action != github.PullRequestActionOpened && pr.Action != github.PullRequestActionReopened && !cpLabelUpdated {
+		return nil
+	}
+
+	var (
+		org    = pr.Repo.Owner.Login
+		repo   = pr.Repo.Name
+		branch = pr.PullRequest.Base.Ref
+	)
+
+	// if the branch doesn't match against the branch names allowed for cherry-picks,
+	// don't do anything
+	if !branchRe.MatchString(branch) {
+		return nil
+	}
+
+	labels, err := gc.GetIssueLabels(org, repo, pr.Number)
+	if err != nil {
+		log.WithError(err).Errorf("failed to list labels")
+		return err
+	}
+	hasCherryPickApprovedLabel := github.HasLabel(cpApprovedLabel, labels)
+	hasCherryPickUnapprovedLabel := github.HasLabel(cpUnapprovedLabel, labels)
+
+	// if it has the approved label,
+	// remove the unapproved label (if it exists) and
+	// remove any comments left by this plugin
+	if hasCherryPickApprovedLabel {
+		if hasCherryPickUnapprovedLabel {
+			if err := gc.RemoveLabel(org, repo, pr.Number, cpUnapprovedLabel); err != nil {
+				log.WithError(err).Errorf("Github failed to remove the following label: %s", cpUnapprovedLabel)
+			}
+		}
+		cp.PruneComments(shouldPrune(log, comment))
+		return nil
+	}
+
+	// if it already has the unapproved label, we are done here
+	if hasCherryPickUnapprovedLabel {
+		return nil
+	}
+
+	// only add the label and comment if none of the approved and unapproved labels are present
+	if err := gc.AddLabel(org, repo, pr.Number, cpUnapprovedLabel); err != nil {
+		log.WithError(err).Errorf("Github failed to add the following label: %s", cpUnapprovedLabel)
+	}
+
+	formattedComment := plugins.FormatSimpleResponse(pr.PullRequest.User.Login, comment)
+	if err := gc.CreateComment(org, repo, pr.Number, formattedComment); err != nil {
+		log.WithError(err).Errorf("Failed to comment %q", formattedComment)
+	}
+
+	return nil
+}
+
+// shouldPrune finds comments left by this plugin.
+func shouldPrune(log *logrus.Entry, commentBody string) func(github.IssueComment) bool {
```

Comment:
`log` is not used.
You could just get rid of this entire function now and pass this value to PruneComments instead:
```go
func(comment github.IssueComment) bool {
	return strings.Contains(comment.Body, comment)
}
```

Reply:
Done.